crypto/bn/bn_recp.c

```diff
@@ -15,6 +15,7 @@ void BN_RECP_CTX_init(BN_RECP_CTX *recp)
     bn_init(&(recp->N));
     bn_init(&(recp->Nr));
     recp->num_bits = 0;
+    recp->shift = 0;
```

Comment:
put the memset before line 15; remove 17 and 18.  geting in the habit of memset for all zeroizing is a good one for openssl i think.